### PR TITLE
[Java] Add JNI bindings for `integers_to_hex`

### LINKED ISSUE #14081: [FEA][JNI] Add JNI bindings for cudf::strings::integers_to_hex()
**Is your feature request related to a problem? Please describe.**
We need a way to convert a string col to a hex string col

**Describe the solution you'd like**
libcudf already supports this in convert_integers::integers_to_hex(). We should expose this method to Java

**Describe alternatives you've considered**
None

### PR DESCRIPTION
**Pull request by razajafri**
## Description
This PR adds a method to ColumnView class to allow for conversion from Integers to hex
closes #14081 

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by kuhushukla — APPROVED** [review]
lgtm

File: java/src/main/java/ai/rapids/cudf/ColumnView.java [hunk 1]
```diff
@@ -5231,4 +5233,29 @@ static ColumnView[] getColumnViewsFromPointers(long[] nativeHandles) {
...
+   * s is [ '04D2', 'FFFFFFFF', '00', '1B', '146D7719']
+   *
+   * The example above shows an `INT32` type column where each integer is 4 bytes.
+   * Leading zeros are suppressed unless filling out a complete byte as in
+   * `123 -> '04D2'` instead of `000004D2` or `4D2`. 
+   * 
+   * @return new string ColumnVector
+   */
+  public ColumnVector toHex() {
+    assert getType().isIntegral() : "Only integers are supported";
```
Comment (revans2) [inline java/src/main/java/ai/rapids/cudf/ColumnView.java:5258]: nit: technically I don't think we need this. The native code should handle checking for this and throw an exception if needed.

**Review by revans2 — APPROVED** [review]

File: java/src/main/java/ai/rapids/cudf/ColumnView.java (hunk 1 above)
Comment (razajafri) [reply java/src/main/java/ai/rapids/cudf/ColumnView.java:5258]: Yes, but the native code is allowing bool8 which will return invalid results.

Comment (revans2) [reply java/src/main/java/ai/rapids/cudf/ColumnView.java:5258]: That is a CUDF bug and there is a PR up to fix it. https://github.com/rapidsai/cudf/pull/14208  Plus the assert will only run if you enable them. They are not on by default.

Comment (razajafri) [reply java/src/main/java/ai/rapids/cudf/ColumnView.java:5258]: I understand that. By convention we have been using asserts, do you want me to revert this entire assert and related change?

Comment (razajafri) [reply java/src/main/java/ai/rapids/cudf/ColumnView.java:5258]: Thank you very much

**Pull request merged**